component/config_test.go

```diff
@@ -194,3 +195,222 @@ func TestValidateConfig(t *testing.T) {
 		})
 	}
 }
+
+func TestNewType(t *testing.T) {
+	tests := []struct {
+		name      string
+		shouldErr bool
+	}{
+		{name: "active_directory_ds"},
```

Comment:
These are all the component types we have today, at least for components that have `metadata.yaml`